src/kinetics/KineticsFactory.cpp

```diff
@@ -168,17 +171,44 @@ void addReactions(Kinetics& kin, const AnyMap& phaseNode, const AnyMap& rootNode
             AnyMap reactions = AnyMap::fromYamlFile(fileName,
                 rootNode.getString("__file__", ""));
             for (const auto& R : reactions[node].asVector<AnyMap>()) {
-                kin.addReaction(newReaction(R, kin));
+                try{
+                    kin.addReaction(newReaction(R, kin));
+                }
+                catch(CanteraError& err){
+                    format_to(err_Plog_reactions, "{}", err.what());
+                }
             }
         } else {
             // specified section is in the current file
             for (const auto& R : rootNode.at(sections[i]).asVector<AnyMap>()) {
-                kin.addReaction(newReaction(R, kin));
+                try {
+                    kin.addReaction(newReaction(R, kin));
+                }
+                catch (CanteraError& err){
+                    std::string captured_err = err.what();
+                    captured_err.erase(std::remove(captured_err.begin(), captured_err.end(), '*'), captured_err.end());
+                    if (captured_err.find("Plog::validate") != std::string::npos) {
+                        captured_err.erase(0, 40);
+                        format_to(err_Plog_reactions, "{}", captured_err);
+                    } else {
+                        std::regex re("(\\w+)::(\\w+)");
+                        std::smatch match;
+                        regex_search(captured_err, match, re);
+                        std::string procedure = match[0];
+                        std::string msg = captured_err.substr(25+procedure.length(), -1);
+                        throw CanteraError(procedure,
+                            "{}", msg);
```

Comment:
Why is this processing necessary? What are you erasing here that wasn't here before? Processing based on character lengths is quite fragile, as I mentioned earlier, since changes in other places can totally break this without any warning.

Reply:
Here I am trying to raise Plog error while not affecting raising other `CanteraError`, so I  distinguish the `Plog::validation` error and other `CanteraError`,  the first erase line(line189) is for erasing all the '*' sign automatically added in each CanteraError. `captured_err.erase(0, 40)` erases string `CanteraError thrown by Plog::validation`. If the captured `CanteraError` is not a `Plog::validation` error, it uses regex to extract the procedure(the function that raise the error) and message in `CanteraError` and use `throw CanteraError` to raise it again. I forgot to do the same in line 178.  In this case, as long as we do not change the `CanteraError` format like `CanteraError thrown by (\\w+)::(\\w+)` and `Plog::validation` in Plog error message, it should be ok. However, I agree it is fragile to process string based on character length or regex, and I would like to take any suggestion about how to improve this part. I think maybe a function to separate `CanteraError` into predure and message would be good?